### PR TITLE
Separate sinon and sandbox visitors

### PR DESCRIPTION
**Pull request by michaeltamaki-okta**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
     visitor: {
+      ImportDeclaration(path) {
+        /**
+         * If importing sinon, remove the declaration and references in the code
+         * - import sinon from "sinon"
+         * - import $sandbox from "sandbox"
+         */
+        if (path.get("source").isStringLiteral({ value: "sinon" })) {
+          const sinonName = path.node.specifiers[0].local.name;
+          path.findParent((parentPath) => parentPath.isProgram()).traverse(removeSinonVisitor(sinonName, t));
```
Comment (TianGan-okta) [inline src/index.js:15]: Nice!

File: src/sinonSandbox.js [hunk 2]
```diff
@@ -1,25 +1,304 @@
...
+    if (path.get("callee").isMemberExpression() && path.get("callee.object").isIdentifier({ name: sinonName })) {
+      // .createSandbox()
+      if (path.get("callee.property").isIdentifier({ name: "createSandbox" })) {
+        // Remove references to the sinon sandbox timer if useFakeTimers: true
+        if (
+          path.get("arguments").length > 0 &&
+          path.get("arguments.0").isObjectExpression() &&
+          path.node.arguments[0].properties.findIndex((prop) => prop.key.name === "useFakeTimers") !== -1
+        ) {
+          path.findParent((parentPath) => parentPath.isProgram()).traverse(convertTimersVisitor("clock", false, t));
```
Comment (TianGan-okta) [inline src/sinonSandbox.js:119]: Why we need "clock" here?

**Review by TianGan-okta — APPROVED** [review]
Great!

File: src/sinonSandbox.js (hunk 2 above)
Comment (michaeltamaki-okta) [reply src/sinonSandbox.js:119]: When `useFakeTimers` is used, then the sandbox will have a property called `clock` that refers to the mock timer

**Comment by michaeltamaki-okta** [comment]
PR is not updating based on the latest branch updates, will open a new one

**Pull request closed without merging**